python/ray/data/tests/test_dataset_binary.py

```diff
@@ -95,7 +95,7 @@ def test_read_binary_snappy(ray_start_regular_shared, tmp_path):
         path,
         arrow_open_stream_args=dict(compression="snappy"),
     )
-    assert sorted(ds.take()) == [byte_str]
+    assert sorted(ds.take()) == [{"bytes": byte_str}]
```

Comment:
I believe if you use TENSOR_COLUMN_NAME then it will unwrap single-column tables, so we wouldn't break backwards compatibility here.

If we do need to break backwards compatibility, I think we also need a way to fallback, or better yet create a new method.

Reply:
ah thanks for pointing it out, noticing the special handling for `TENSOR_COLUMN_NAME` [here](https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/table_block.py#L152-L155). I am good with setting the column name to be `TENSOR_COLUMN_NAME` to keep backwards compatibility of `binary_ds.take()`, TBH "bytes" vs "__value__" not matter much for users, I think.

Reply:
cc @clarkzinzow WDYT?

Reply:
Tried with `TENSOR_COLUMN_NAME`, but found it does not work because with `TENSOR_COLUMN_NAME` ([exception stack trace](https://gist.github.com/c21/2f300478398d9825f16ccf9932587360)), we are expecting the column to be a NumPy ndarray - https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/arrow_block.py#L117-L122 . 

I gues we have to break backwards compatiblity here. cc @clarkzinzow as well.

Reply:
I don't think that we'd want to reuse `TENSOR_COLUMN_NAME` here unless we expanded the scope of it a good bit, `TENSOR_COLUMN_NAME` is very much tied to the data being a tensor, which we can't assume for `read_binary_files()`. We were going to previously generalize this to `VALUE_COLUMN_NAME` and represent simple blocks as a single-column table, where we'd use a binary extension type that pickles the data if it can't be represented natively in Arrow, but we deprioritized this once it proved to be a good bit of work.

Why don't we just use the existing `"bytes"` column name, where the user-facing representation is now a single-bytes-column table? If the user wants to read binary data and decode it in a UDF, that column name/representation seems reasonable.

Reply:
> Why don't we just use the existing "bytes" column name, where the user-facing representation for read_binary_files is now a single-bytes-column table?

I think the main reason is `binary_ds.take()` returns a list of dict [{"bytes": xxx}, {"bytes": yyy}], instead of list of binary. So user's existing code will break and need code change on user side.

Would that be a huge problem if we break the backwards compatbility here? @ericl

Reply:
In general I don't think we should break backwards compatibility lightly. I would strongly prefer we try to preserve this behavior, or make a new method and deprecate the old one.

Reply:
Cool, let me think more about how to resolve it.

Reply:
@ericl, @clarkzinzow - changed code to keep the same behavior for unwraping single-column binary data source. could you guys take a look of approach? Thanks.